nfs.py

```diff
@@ -44,7 +45,14 @@ def _add(self, dir_name: str) -> None:
 
     def _export_fs(self) -> None:
         self._host.run("systemctl enable nfs-server")
-        self._host.run_or_die("systemctl restart nfs-server")
```

Comment:
Can you explain why it fails? I never encountered this issue.

Reply:
no, I don't know.

Happens very frequently to me. Just twice in a row on `wsfd-advnetlab228.anl.eng.bos2.dc.redhat.com`. ... each time running more than an hour before failing the script.

It must have also happend to @SalDaniele as he did fbf00fa192c61c4100c4fc9d731488193a881271 ? But that  is no solution, because it merely means the script fails earlier before encountering another issue.

Reply:
Yes I believe I hit this issue due to both workers trying to write /etc/exports at the same time. If NFS is misconfigured, we should bail out at that point, since the Redfish boot will fail.

Reply:
I am curious why you are hitting the issue so frequently, I have not encountered this. Are you able to recover NFS between runs somehow?

Either way, adding a lock makes sense to me here.

Reply:
> I am curious why you are hitting the issue so frequently, I have not encountered this.

I don't know.

Why did you then do fbf00fa ?

Reply:
I have not encountered this = encountered this at this frequency.

We need fbf00fa because if NFS fails CDA should report this at this will cause the deployment to fail when we try to iso boot the server. If we don't bail on an NFS restart error, then it will not be obvious why the iso boot failed, as you found firsthand.